backend/core/users/services.py

```
@@ -0,0 +1,54 @@
+	"""Register a new user account and return API key credentials."""
+	engine = get_engine()
+	if engine is None:
+		raise ValidationError("Database engine is not initialized")
+
+	normalized_email = validate_email(email)
+	validated_password = validate_password(password)
+
+	with Session(engine) as session:
+		existing = session.exec(select(User).where(User.email == normalized_email)).first()
+		if existing is not None:
+			raise ValidationError("Email already registered")
```

Ruff (0.15.4)

[warning] Avoid specifying long messages outside the exception class

(TRY003)